app/routers/calendar_grid.py

```diff
@@ -0,0 +1,197 @@
+import calendar
+import itertools
+import locale
+from datetime import date, datetime, timedelta
+from typing import Dict, Iterator, List, Tuple
+
+MONTH_BLOCK: int = 6
+
+locale.setlocale(locale.LC_TIME, ("en", "UTF-8"))
+
+
+class Day:
+    """A Day class.
+
+    Args:
+        date    (datetime): A single datetime date.
+    Arguments:
+        date    (datetime): A single datetime date.
+        sday         (str): The day name.
+        dailyevents (List): List of tuples represent daily event information.
+        events      (List): List of tuples represent time event name.
+                            EX: [("09AP", "Meeting with yam")]
+        css         (Dict): All css classes represent day.
+    """
+
+    def __init__(self, date: datetime):
+        self.date: datetime = date
+        self.sday: str = self.date.strftime("%A")
+        self.dailyevents: List[Tuple] = []
+        self.events:  List[Tuple] = []
+        self.css: Dict[str, str] = {
+            'div': 'day',
+            'date': 'day-number',
+            'daily_event': 'month-event',
+            'daily_event_front': ' '.join([
+                'daily',
+                'front',
+                'background-warmyellow'
+            ]),
+            'daily_event_back': ' '.join([
+                'daily',
+                'back',
+                'text-darkblue',
+                'background-lightgray'
+            ]),
+            'event': 'event',
+        }
+
+    def __str__(self) -> str:
+        return self.date.strftime("%d")
+
+    def display(self) -> str:
+        """Returns day date inf the format of 00 MONTH 00"""
+        return self.date.strftime("%d %B %y").upper()
+
+    @ classmethod
+    def convert_str_to_date(cls, date_string: str) -> datetime:
+        return datetime.strptime(date_string, '%d %B %y')
+
+    @ classmethod
+    def is_weekend(cls, date: date) -> bool:
+        """Returns true if this day is represent a weekend."""
+        return date.strftime("%A") in Week.DAYS_OF_THE_WEEK[-2:]
+
+
+class Week:
+    WEEK_DAYS: int = 7
+    DAYS_OF_THE_WEEK: List[str] = calendar.day_name
+
+    def __init__(self, days: List[Day]):
+        self.days: List[Day] = days
+
+
+class DayWeekend(Day):
+    def __init__(self, date: datetime):
+        super().__init__(date)
+        self.css = {
+            'div': 'day ',
```

Comment:
What does this field mean?

Reply:
The day div - I changes it to a more understandable name.